ts/components/helpers/withConditionalView.tsx

```diff
@@ -1,21 +1,32 @@
-/**
- * A HOC to display the WrappedComponent when the check function is verified , otherwise the ConditionComponent will be displayed
- */
-
 import hoistNonReactStatics from "hoist-non-react-statics";
 import React from "react";
+import { NavigationInjectedProps } from "react-navigation";
 
+/**
+ * A HOC to display the WrappedComponent when the check function is verified,
+ * otherwise the ConditionComponent will be displayed
+ */
 export function withConditionalView<P, T, C>(
   WrappedComponent: React.ComponentType<P>,
   check: (props: T) => boolean,
   ConditionComponent: React.ComponentType<C>
 ) {
-  class ConditionalView extends React.PureComponent {
+  /**
+   * The component should inherit the navigation params of both the wrapped and
+   * the contition component to ensure the nagivigation can address the proper
+   * navigation paramaters
+   */
+  type nullN = {};
+  type NP = P extends NavigationInjectedProps<infer N> ? N : nullN;
+  type NC = C extends NavigationInjectedProps<infer N> ? N : nullN;
```

Comment:
```suggestion
  type NC = C extends NavigationInjectedProps<infer N> ? N : never;
```

Reply:
it works! updated within https://github.com/pagopa/io-app/pull/1668/commits/8494b7785a0e25e51e0791ee94f868920dfa1c24

Reply:
it works! updated within https://github.com/pagopa/io-app/pull/1668/commits/8494b7785a0e25e51e0791ee94f868920dfa1c24

Reply:
same above